roles/os_firewall/tasks/iptables.yml

```diff
@@ -4,17 +4,28 @@
   systemd:
     name: firewalld
     state: stopped
-    enabled: no
-    masked: yes
   register: task_result
   failed_when:
     - task_result is failed
     - ('could not' not in task_result.msg|lower)
 
+# These tasks are split in two, as changing masked and enabled status
+# would mutate `changed` field, so it needs to be performed in two tasks
+- name: Mask and disable firewalld service
+  systemd:
+    name: firewalld
+    enabled: no
+    masked: yes
+  when:
+  - task_result is succeeded
```

Comment:
Can both of these be true?  Does that cover all cases?

I think we should remove the when condition and just use the same failed_when condition as above.